pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -1538,7 +1538,7 @@ func requiresInternalLoadBalancer(service *v1.Service) bool {
 
 func subnet(service *v1.Service) *string {
 	if requiresInternalLoadBalancer(service) {
-		if l, found := service.Annotations[ServiceAnnotationLoadBalancerInternalSubnet]; found {
+		if l, found := service.Annotations[ServiceAnnotationLoadBalancerInternalSubnet]; found && strings.TrimSpace(l) != "" {
```

Comment:
if it returns pointer to "", code will jump to following, and then still return error:
https://github.com/kubernetes/kubernetes/blob/8d0c5d9727d458da51e40810387a3d9e2030fd68/pkg/cloudprovider/providers/azure/azure_loadbalancer.go#L532

Reply:
it won't error if this case, because `nil` would be returned.